thetis/solver2d.py

```diff
@@ -297,6 +297,18 @@ def create_function_spaces(self):
 
         self._isfrozen = True
 
+    def add_new_field(self, label):
```

Comment:
I guess I was thinking about something more generic such as
```python
def add_new_field(self, function, label, name, filename, shortname=None, unit='-'):
    metadata[label] = {
        'name': name, ...
    }
    field_metadata[label] = metadata
    self.fields[label] = function
```
That way users could add any e.g. diagnostic function and it could be exported via `options.fields_to_export`. In the special case of tracers you would just populate the args from `options.tracer_metadata`.